mlflow/utils/search_utils.py

```diff
@@ -317,30 +318,43 @@ def run_matches(run):
         return [run for run in runs if run_matches(run)]
 
     @classmethod
-    def parse_order_by(cls, order_by):
+    def _parse_order_by_string(cls, order_by):
         try:
             parsed = sqlparse.parse(order_by)
         except Exception:
-            raise MlflowException("Error on parsing order_by clause '%s'" % order_by,
+            raise MlflowException(f"Error on parsing order_by clause '{order_by}'",
                                   error_code=INVALID_PARAMETER_VALUE)
         if len(parsed) != 1 or not isinstance(parsed[0], Statement):
-            raise MlflowException("Invalid order_by clause '%s'. Could not be parsed." %
-                                  order_by, error_code=INVALID_PARAMETER_VALUE)
-
+            raise MlflowException(f"Invalid order_by clause '{order_by}'. Could not be parsed.",
+                                  error_code=INVALID_PARAMETER_VALUE)
         statement = parsed[0]
         if len(statement.tokens) != 1 or not isinstance(statement[0], Identifier):
-            raise MlflowException("Invalid order_by clause '%s'. Could not be parsed." %
-                                  order_by, error_code=INVALID_PARAMETER_VALUE)
-
+            raise MlflowException(f"Invalid order_by clause '{order_by}'. Could not be parsed.",
+                                  error_code=INVALID_PARAMETER_VALUE)
         token_value = statement.tokens[0].value
         is_ascending = True
         if token_value.lower().endswith(" desc"):
             is_ascending = False
             token_value = token_value[0:-len(" desc")]
         elif token_value.lower().endswith(" asc"):
             token_value = token_value[0:-len(" asc")]
+        return token_value, is_ascending
+
+    @classmethod
+    def parse_order_by_runs(cls, order_by):
+        token_value, is_ascending = cls._parse_order_by_string(order_by)
         identifier = cls._get_identifier(token_value.strip(), cls.VALID_ORDER_BY_ATTRIBUTE_KEYS)
-        return (identifier["type"], identifier["key"], is_ascending)
+        return identifier["type"], identifier["key"], is_ascending
+
+    @classmethod
+    def parse_order_by_registered_models(cls, order_by):
```

Comment:
rename `parse_order_by_for_search_registered_models`
at some point soon (different pull request), we should refactor this as `SearchUtils` base class simpler/shorter method names and derived classes for `SearchRuns`, `SearchModelVersions`, `SearchRegisteredModels`, ... etc.

Reply:
Yeah definitely agree